Assets/Scripts/Missions/MissionManager.cs

```diff
@@ -125,55 +105,46 @@ public StampCollectible GetStampCollectibleFromEnum(MissionsEnum missionEnumValu
-        GameObject createdMission = Instantiate(missionPrefab, Vector3.zero, Quaternion.identity, transform);
-        Component missionComponent = createdMission.GetComponent(typeof(AMission));
+        AMission missionLogic = GetMissionLogic(missionEnumValue);
+        missionLogic.gameObject.SetActive(true);
 
-        if (missionComponent is AMission)
-        {
-            AMission mission = missionComponent as AMission;
-            SetObjectiveTextForList(missionEnumValue);
-
-            SetInstantiatedMissionForEnum(missionEnumValue, mission);
-            ProgressManager.Instance.AddMission(mission);
-            activeMissions.Add(new InProgressMissionContainer(mission, createdMission));
-            return mission;
-        } 
-        else
-        {
-            Utils.LogErrorAndStopPlayMode($"Expected to find an AMission component on {missionPrefab.name}!");
-            return null;
-        }
+        ProgressManager.Instance.AddMission(missionLogic);
+        activeMissions.Add(missionEnumValue);
+
+        SetObjectiveTextForList(missionEnumValue);
+        
+        return missionLogic;
     }
 
-    public InProgressMissionContainer GetActiveMission<T>()
+    public bool IsMissionActive(MissionsEnum missionEnumValue)
     {
-        return activeMissions.Find(m => m.mission is T);
+        return activeMissions.Contains(missionEnumValue);
     }
 
     public void RestartMission(MissionsEnum missionEnumValue)
     {
-        ProgressManager.Instance.UpdateMissionStatus(GetInstantiatedMissionFromEnum(missionEnumValue), MissionStatusCode.Started);
+        ProgressManager.Instance.UpdateMissionStatus(GetMissionLogic(missionEnumValue), MissionStatusCode.Started);
     }
 
     public void CompleteMissionObjective(MissionsEnum missionEnumValue)
     {
-        ProgressManager.Instance.UpdateMissionStatus(GetInstantiatedMissionFromEnum(missionEnumValue), MissionStatusCode.Completed);
+        ProgressManager.Instance.UpdateMissionStatus(GetMissionLogic(missionEnumValue), MissionStatusCode.Completed);
         ObjectiveList.Instance.CrossOutObjectiveText();
         Debug.Log("Objective Complete");
     }
 
     public void EndMission(MissionsEnum missionEnumValue)
     {
-        AMission mission = GetInstantiatedMissionFromEnum(missionEnumValue);
-        InProgressMissionContainer container = activeMissions.Find(m => m.mission == mission);
+        AMission missionLogic = GetMissionLogic(missionEnumValue);
+
         ObjectiveList.Instance.HideObjectiveList();
 
-        if (container != null)
+        if (activeMissions.Contains(missionEnumValue))
         {
-            Destroy(container.gameObject);
-            activeMissions.Remove(container);
-            ProgressManager.Instance.UpdateMissionStatus(mission, MissionStatusCode.Closed);
+            activeMissions.Remove(missionEnumValue);
+
+            missionLogic.gameObject.SetActive(false);
```

Comment:
Nice, I like the readability (both in code and in the inspector) of enabling/disabling GameObjects!

Reply:
Yea I think so too! Much clearer now overall :)